server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -401,7 +402,9 @@ private void recoverFromTranslogInternal() throws IOException {
         try (Translog.Snapshot snapshot = translog.newSnapshotFromGen(translogGen)) {
             opsRecovered = config().getTranslogRecoveryRunner().run(this, snapshot);
         } catch (Exception e) {
-            throw new EngineException(shardId, "failed to recover from translog", e);
+            throw new EngineException(shardId, "Failed to recover from translog. Translog path " + translog.location() + ". Consider " +
+                    "running elasticsearch-translog tool",
```

Comment:
I wouldn't recommend running the corruption tool. It's a dangerous one and need much more context and explanation. We should leave it to the docs. Also, the exception should coming out of the translog should indicate what it is and if it's a corruption should include a path.